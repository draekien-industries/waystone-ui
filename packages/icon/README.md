# `@waystone/icon`

## Installation

Run the relevant script for your package manager

| Package manager | Script                                                 |
| --------------- | ------------------------------------------------------ |
| npm             | `npm i @waystone/ui@latest @emotion/react theme-ui`    |
| yarn            | `yarn add @waystone/ui@latest @emotion/react theme-ui` |
| pnpm            | `pnpm i @waystone/ui@latest @emotion/react theme-ui`   |

## Usage


