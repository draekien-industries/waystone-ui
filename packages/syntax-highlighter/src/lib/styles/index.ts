export * from './a11yDark.generated';
export * from './a11yLight.generated';
export * from './agate.generated';
export * from './anOldHope.generated';
export * from './androidstudio.generated';
export * from './arduinoLight.generated';
export * from './arta.generated';
export * from './ascetic.generated';
export * from './atomOneDarkReasonable.generated';
export * from './atomOneDark.generated';
export * from './atomOneLight.generated';
export * from './brownPaper.generated';
export * from './codepenEmbed.generated';
export * from './colorBrewer.generated';
export * from './dark.generated';
export * from './defaultStyle.generated';
export * from './devibeans.generated';
export * from './docco.generated';
export * from './far.generated';
export * from './felipec.generated';
export * from './foundation.generated';
export * from './githubDarkDimmed.generated';
export * from './githubDark.generated';
export * from './github.generated';
export * from './gml.generated';
export * from './googlecode.generated';
export * from './gradientDark.generated';
export * from './gradientLight.generated';
export * from './grayscale.generated';
export * from './hybrid.generated';
export * from './idea.generated';
export * from './intellijLight.generated';
export * from './irBlack.generated';
export * from './isblEditorDark.generated';
export * from './isblEditorLight.generated';
export * from './kimbieDark.generated';
export * from './kimbieLight.generated';
export * from './lightfair.generated';
export * from './lioshi.generated';
export * from './magula.generated';
export * from './monoBlue.generated';
export * from './monokaiSublime.generated';
export * from './monokai.generated';
export * from './nightOwl.generated';
export * from './nnfxDark.generated';
export * from './nnfxLight.generated';
export * from './nord.generated';
export * from './obsidian.generated';
export * from './pandaSyntaxDark.generated';
export * from './pandaSyntaxLight.generated';
export * from './paraisoDark.generated';
export * from './paraisoLight.generated';
export * from './pojoaque.generated';
export * from './purebasic.generated';
export * from './qtcreatorDark.generated';
export * from './qtcreatorLight.generated';
export * from './rainbow.generated';
export * from './routeros.generated';
export * from './schoolBook.generated';
export * from './shadesOfPurple.generated';
export * from './srcery.generated';
export * from './stackoverflowDark.generated';
export * from './stackoverflowLight.generated';
export * from './sunburst.generated';
export * from './tokyoNightDark.generated';
export * from './tokyoNightLight.generated';
export * from './tomorrowNightBlue.generated';
export * from './tomorrowNightBright.generated';
export * from './vs.generated';
export * from './vs2015.generated';
export * from './xcode.generated';
export * from './xt256.generated';
