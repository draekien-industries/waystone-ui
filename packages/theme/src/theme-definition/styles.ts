export const styles = {
  shadows: {
    none: 'none',
    sm: 'inset 0 0 0.5px 1px hsla(0, 0%, 100%, 0.075), 0 0 0 1px hsla(0, 0%, 0%, 0.05), 0 0.3px 0.4px hsla(0, 0%, 0%, 0.02), 0 0.9px 1.5px hsla(0, 0%, 0%, 0.045), 0 3.5px 6px hsla(0, 0%, 0%, 0.09)',
    md: 'inset 0 0 0.5px 1px hsla(0, 0%, 100%, 0.075), 0 0 4px 1px hsla(0, 0%, 0%, 0.05), 0 0.3px 0.4px hsla(0, 0%, 0%, 0.02), 0 0.9px 1.5px hsla(0, 0%, 0%, 0.045), 0 3.5px 6px hsla(0, 0%, 0%, 0.09)',
    lg: 'inset 0 0 0.5px 1px hsla(0, 0%, 100%, 0.075), 0 0 8px 1px hsla(0, 0%, 0%, 0.05), 0 0.3px 0.4px hsla(0, 0%, 0%, 0.02), 0 0.9px 1.5px hsla(0, 0%, 0%, 0.045), 0 3.5px 6px hsla(0, 0%, 0%, 0.09)',
  },
  sizes: {
    xs: '1em',
    sm: '5em',
    md: '10em',
    lg: '15em',
    xl: '20em',
    half: '50%',
    full: '100%',
  },
} as const;

export default styles;
