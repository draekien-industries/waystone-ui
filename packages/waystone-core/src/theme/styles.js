export const styles = {
    shadows: {
        sm: '0 0 0.25em 0 rgba(4, 40, 52, 0.1), 0 0.125em 0.25em -0.125em rgba(4, 40, 52, 0.25)',
        md: '0 0 0.25em 0 rgba(4, 40, 52, 0.1), 0 0.5em 1em -0.5em rgba(4, 40, 52, 0.25)',
        lg: '0 0 0.5em 0 rgba(4, 40, 52, 0.1), 0 1em 2em -1em rgba(3, 36, 48, 0.25)',
        'sm-dark': '0 0 0.25em 0 #1b1b1bFF, 0 0.125em 0.25em -0.125em #0b0b0bA0',
        'md-dark': '0 0 0.25em 0 #1b1b1bFF, 0 0.5em 1em -0.5em #0b0b0bA0',
        'lg-dark': '0 0 0.5em 0 #1b1b1bFF, 0 1em 2em -1em #0b0b0bA0',
    },
    sizes: {
        xs: '1em',
        sm: '5em',
        md: '10em',
        lg: '15em',
        xl: '20em',
        half: '50%',
        full: '100%',
    },
};
export default styles;
