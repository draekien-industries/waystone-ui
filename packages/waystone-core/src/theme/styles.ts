export const styles = {
  shadows: {
    sm: '0 0 0.25em 0 rgba(4, 40, 52, 0.1), 0 0.125em 0.25em -0.125em rgba(4, 40, 52, 0.25)',
    md: '0 0 0.25em 0 rgba(4, 40, 52, 0.1), 0 0.5em 1em -0.5em rgba(4, 40, 52, 0.25)',
    lg: '0 0 0.5em 0 rgba(4, 40, 52, 0.1), 0 1em 2em -1em rgba(3, 36, 48, 0.25)',
    'sm-dark':
      '0 0 0.5em 0 rgba(27, 27, 27, 1), 0 0.5 1em -0.5em rgba(11, 11, 11, 0.73)',
    'md-dark':
      '0 0 1em 0 rgba(27, 27, 27, 1), 0 1em 2em -1em rgba(11, 11, 11, 0.73)',
    'lg-dark':
      '0 0 2em 0 rgba(27, 27, 27, 1), 0 2em 4em -2em rgba(11, 11, 11, 0.73)',
  },
  sizes: {
    xs: '1em',
    sm: '5em',
    md: '10em',
    lg: '15em',
    xl: '20em',
    half: '50%',
    full: '100%',
  },
};

export default styles;
