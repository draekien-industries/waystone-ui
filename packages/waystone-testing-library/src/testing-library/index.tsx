'use client';

import type { RenderOptions, RenderResult } from '@testing-library/react';
import { render } from '@testing-library/react';
import { Waystone } from '@waystone/theme-provider';
import type { PropsWithChildren } from 'react';
import type React from 'react';
import { useEffect } from 'react';
import { useColorMode } from 'theme-ui';

const TestProvider = ({ children }: PropsWithChildren) => (
  <Waystone>{children}</Waystone>
);

const DarkModeWrapper = ({ children }: PropsWithChildren) => {
  const [, setColorMode] = useColorMode();

  useEffect(() => {
    setColorMode('dark');
  }, [setColorMode]);

  return <div>{children}</div>;
};

const themedRender: (
  ui: React.ReactElement,
  options?: RenderOptions
) => RenderResult = (ui, options?) =>
  render(ui, { wrapper: options?.wrapper || TestProvider, ...options });

export * from '@testing-library/react';
export * from '@testing-library/user-event';

export { DarkModeWrapper, themedRender as render };
