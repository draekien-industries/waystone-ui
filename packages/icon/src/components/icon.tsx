import { iconCss } from './icon.styles';
import type { IconProps } from './icon.types';
import { getFontSize, getIconClassName } from './icon.utils';

export const Icon = ({
  children,
  color = 'inherit',
  size = 'small',
  variant = 'outlined',
  className,
  ...rest
}: IconProps) => {
  const fontSize = getFontSize(size);
  const iconClassName = getIconClassName(variant, className);

  return (
    <i className={iconClassName} {...rest} sx={iconCss({ color, fontSize })}>
      {children}
    </i>
  );
};
