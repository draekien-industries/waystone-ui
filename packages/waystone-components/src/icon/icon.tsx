/** @jsxImportSource theme-ui */
import React from 'react';
import { getIconClassName } from './icon.fx';
import { iconCss } from './icon.styles';

import { HasColor, HasSize, HasVariant } from '../common/interfaces';

export type IconVariant =
  | 'filled'
  | 'outlined'
  | 'round'
  | 'two-tone'
  | 'sharp';

/** The props required to generate the correct icon styles. */
export interface IconStyleProps extends HasSize, HasColor {}

/**
 * The options that can be set when rendering the icon.
 * @extends IconStyleProps
 * @extends HasVariant
 */
export interface IconOptions extends IconStyleProps, HasVariant {
  /**
   * The variant of icon to render.
   * @override
   */
  variant?: IconVariant;
}

/** Props for the icon component. */
export interface IconProps
  extends Omit<React.HTMLAttributes<HTMLElement>, 'children' | 'color'>,
    HasVariant,
    HasSize,
    HasColor {
  /** The name of the material icon to render. */
  name: string;
  /** The material icon variant to use when rendering the icon. */
  variant?: IconVariant;
}

/**
 * Renders an icon component.
 * This component library relies on the material-icons font for rendering icons.
 * You can view the list of available icons by going to https://fonts.google.com/icons
 */
export const Icon = ({
  className,
  name,
  variant = 'filled',
  size = 'md',
  color = 'inherit',
  ...rest
}: IconProps) => {
  return (
    <i
      sx={iconCss({ size, color })}
      className={getIconClassName(variant, className)}
      {...rest}>
      {name}
    </i>
  );
};
