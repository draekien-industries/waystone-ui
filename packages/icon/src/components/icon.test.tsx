import { render } from '@waystone/testing-library';
import { Icon } from './icon';
import type { IconSize, IconVariant } from './icon.types';
import { iconCss } from './icon.styles';

describe('<Icon />', () => {
  const sizes = [
    'large',
    'medium',
    'small',
    'extra-small',
  ] as const satisfies IconSize[];

  const variants = [
    'filled',
    'outlined',
    'round',
    'sharp',
    'two-tone',
  ] as const satisfies IconVariant[];

  describe('component', () => {
    it('should match snapshot', () => {
      const { container } = render(<Icon>accessibility</Icon>);

      expect(container).toMatchSnapshot();
    });

    describe.each(sizes)('size %s', (size) => {
      it('should match snapshot', () => {
        const { container } = render(<Icon size={size}>accessibility</Icon>);

        expect(container).toMatchSnapshot();
      });
    });

    describe.each(variants)('variant %s', (variant) => {
      it('should match snapshot', () => {
        const { container } = render(
          <Icon variant={variant}>accessibility</Icon>
        );

        expect(container).toMatchSnapshot();
      });
    });
  });

  describe('css', () => {
    it('should match snapshot', () => {
      const css = iconCss({});

      expect(css).toMatchSnapshot();
    });
  });
});
