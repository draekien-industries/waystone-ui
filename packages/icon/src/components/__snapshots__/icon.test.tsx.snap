// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Icon /> component should match snapshot 1`] = `
<div>
  <i
    class="material-icons-outlined css-1ye7pjf-Icon"
  >
    accessibility
  </i>
</div>
`;

exports[`<Icon /> component size extra-small should match snapshot 1`] = `
<div>
  <i
    class="material-icons-outlined css-4jdh4r-Icon"
  >
    accessibility
  </i>
</div>
`;

exports[`<Icon /> component size large should match snapshot 1`] = `
<div>
  <i
    class="material-icons-outlined css-pj8p8w-Icon"
  >
    accessibility
  </i>
</div>
`;

exports[`<Icon /> component size medium should match snapshot 1`] = `
<div>
  <i
    class="material-icons-outlined css-1h9809z-Icon"
  >
    accessibility
  </i>
</div>
`;

exports[`<Icon /> component size small should match snapshot 1`] = `
<div>
  <i
    class="material-icons-outlined css-1ye7pjf-Icon"
  >
    accessibility
  </i>
</div>
`;

exports[`<Icon /> component variant filled should match snapshot 1`] = `
<div>
  <i
    class="material-icons css-1ye7pjf-Icon"
  >
    accessibility
  </i>
</div>
`;

exports[`<Icon /> component variant outlined should match snapshot 1`] = `
<div>
  <i
    class="material-icons-outlined css-1ye7pjf-Icon"
  >
    accessibility
  </i>
</div>
`;

exports[`<Icon /> component variant round should match snapshot 1`] = `
<div>
  <i
    class="material-icons-round css-1ye7pjf-Icon"
  >
    accessibility
  </i>
</div>
`;

exports[`<Icon /> component variant sharp should match snapshot 1`] = `
<div>
  <i
    class="material-icons-sharp css-1ye7pjf-Icon"
  >
    accessibility
  </i>
</div>
`;

exports[`<Icon /> component variant two-tone should match snapshot 1`] = `
<div>
  <i
    class="material-icons-two-tone css-1ye7pjf-Icon"
  >
    accessibility
  </i>
</div>
`;

exports[`<Icon /> css should match snapshot 1`] = `
{
  "color": undefined,
  "fontSize": undefined,
  "verticalAlign": "middle",
}
`;
